jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -1007,16 +1009,25 @@ def f_jax(x):  # x: f32[w, h]
                      arg_descriptors=[RandArg((3, 4), _f32)],
                      poly_axes=[(0, 1)])
 
-  def test_non_trivial_polynomials_spec(self):
-    if config.jax_dynamic_shapes:
-      raise unittest.SkipTest("--jax_dynamic_shapes supports only trivial polynomials")
+  @parameterized.named_parameters(
+      dict(testcase_name=f"_{str(polymorphic_shapes)}",
+           polymorphic_shapes=polymorphic_shapes)
+      # The polymorphic_shapes should have three comma-separated DimExpr matching
+      # 16, 24, 32
+      for polymorphic_shapes in [
+          "b1+6,b1+14,b2",  # b1=10, b2=32
```

Comment:
What happen if we use extreme value here. e.g "b1+17" so the b1=-1 ?  Does our code handle this exception correctly ?

Reply:
Not yet, there is some code to check that all dimension variables are >= 1, but it only works in very few cases, e.g., in TF eager mode in non-native serialization. This is a known problem: 

jax/experimental/jax2tf/README.md#unchecked-assumption-that-the-dimension-variables-take-strictly-positive-values

This will be checked in native serialization, but I need a few more changes here and in XlaCallModule before we can do that.